\documentclass{bioinfo}
\copyrightyear{2016} \pubyear{2017}

\access{Advance Access Publication Date: Day Month Year}
\appnotes{Application Note}

%\usepackage{tikz}
\usepackage{nicefrac}
\usepackage{comment}
\usepackage{graphicx}

\newcommand{\code}[1]{{\tt #1}}

\begin{document}
\firstpage{1}

\subtitle{}

\title[node.dating]{node.dating: Timing ancestral dates in phylogenetic trees}
\author[Jones \textit{et~al}.]{Bradley R. Jones\,$^{\text{\sfb 1, 2,}*}$ and Art F. Y. Poon\,$^{\text{\sfb 2,3}}$}
\address{$^{\text{\sf 1}}$Faculty of Health Sciences, Simon Fraser University, Burnaby, V5A 1S6, Canada, \\
$^{\text{\sf 2}}$BC Centre for Excellence in HIV/AIDS, Vancouver, V6Z 1Y6, Canada and \\
$^{\text{\sf 2}}$Department of Medicine, University of British Columbia, V5Z 1M9, Canada}

\corresp{$^\ast$To whom correspondence should be addressed.}

\history{Received on XXXXX; revised on XXXXX; accepted on XXXXX}

\editor{Associate Editor: XXXXXXX}

\abstract{
\textbf{Summary:}
(Brief Motivation).
(Phylogenetics to time trees).
Here we present the $R$ software \code{node.dating}, which uses a maximum likelihood method to date the internal nodes of a phylogenetic tree. \\
\textbf{Availability and Implementation:} \code{node.dating} is written in $R$ and requires the $R$ package, \emph{APE}. \code{node.dating} is available () \\
\textbf{Contact:} \href{brj1@sfu.ca}{brj1@sfu.ca}\\
\textbf{Supplementary information:} Supplementary data are available at \textit{Bioinformatics} online.}

\maketitle

%Keywords: 
%Phylogenetics, Ancestor Dating, Linear Regression, Molecular Clock \\

\underline{}
\section{Introduction} \label{sec:intro}
(Motivation.)

Genetic sequences from the same or different species can be organized in phylogenetic trees whose edges record the genetic difference between each sequence.
(Ways to make trees \citep{Raxml14}).
In the presence of a molecular clock, the edges of a phylogenetic tree can be scaled over time to give a `time tree'.
%The nodes of a time tree also carry a time value which for the tips of the tree represent the sampling time of the tip's sequence.
The internal nodes of a time tree represent when the lineages of the tree separated and are the most recent common ancestors (MRCA) of the tips of the tree \citep{}.
The date of MRCA's are of interest to paleogeneticists \citep{} and also in epidemiology where the time of the MRCA can be used as a proxy for the time of infection between patients in an outbreak \citep{}. Recently, \cite{Kumar16} compiled a history of ancestral dating techniques.

A multitude of software has been developed to date MRCA's and create time trees using various techniques such as: linear regression \citep{Tempest}, maximum likelihood \citep{TipDates, r8ts, PAML}, Bayesian analysis \citep{BEAST}, heuristics \citep{UPGMA, TREBLE}, and least squares \citep{LSD}.
(Transition.)
Our software, \code{node.dating}, uses a maximum likelihood approach to date the internal nodes of a phylogenetic tree.
\code{node.dating} is written in \emph{R} and is compatible with the \emph{R} package \emph{APE} \citep{APE}.


\section{Algorithm} \label{sec:alg}
We start with a rooted phylogenetic tree with edge length equal to genetic distance and the dates when each tip of the tree was sampled.

A simple linear regression is used to estimate the mutation rate assuming a strict molecular clock.

To estimate the dates of the internal nodes, we follow an approach given by \cite{Felsenstein81} and inspired by \emph{TipDates} \citep{TipDates}.
A maximum likelihood method is applied locally to date each internal node.
Then using these estimates, the algorithm iterates until a sufficiently approximate maximum likelihood solution is obtained.

\section{Comparison with TempEst and LSD} \label{sec:tempest}
In order to test the viability of \code{node.dating}, we simulated 50 phylogenetic trees and compared the results of \code{node.dating} with \emph{TempEst}'s (formerly \emph{Path-O-Gen}) internal node dating and also with \emph{LSD}.
\emph{TempEst} uses the prediction given by a linear regression to estimate the dates of the internal nodes and \emph{LSD} uses a least squares method.
To compare the results of the different dating software we used a weighted root mean squared error (RMSE).
The details of the simulation method and weighted RMSE are detailed in the Supplementary Information.

We ran a modified version of \emph{TempEst} that converts a phylogenetic tree to a time tree on each of 50 simulated phylogenetic trees and we did the same with \emph{LSD}.
The average weighted RMSE of the MRCA using \emph{TempEst} was $132.$ days and using \emph{LSD} was $24.9$ days.
These are higher than the average weighted RMSE of the MRCA using \code{node.dating}, which was $22.6$ days, though the weighted RMSE of \emph{LSD} is comparable to the RMSE of \code{node.dating}.
Overall, the RMSE of each tree using \code{node.dating} was less than the RMSE using \emph{TempEst}, suggesting that \code{node.dating} is more accurate at dating the internal nodes of phylogenetic trees.
% BEAST 20.0 days

The main purpose of \emph{TempEst} is not to date internal nodes, but to detect/verify the presence of a strict molecular clock.
We did not use \emph{TempEst} or \emph{LSD}'s root-to-tip regression in our experiments because we merely wanted to compare the internal node dating.

\section{Visualization}
In this section, we exhibit a visualization of sequence data using the estimated dates of internal nodes.

We retrieved intra-host patient-derivied sequences from Patient 16617 on the LANL HIV database \citep{LosAlamos}.
This patient's sequence data was collected in \cite{Llewellyn06} as Patient 1180.

The sequences were aligned using MUSCLE 3.8.31 \citep{Muscle04} and inspected and cleaned using AliView \citep{AliView14}. 
We trimmed the alignments so that each sequence had at least 50\% coverage over each base.
We reconstructed the phylogeny of the patient's sequences using \emph{RAxML} 8.2.4 \citep{Raxml14} and rooted the tree using the \code{rtt} function of APE.

We then estimated the dates of the internal nodes using \code{node.dating}. Finally we plotted the genetic distance from the root versus time of the internal nodes and the sampled sequences. This plot is displayed in Figure \ref{fig:pat16617}.

\begin{figure}[t]
	\centering
	\includegraphics[width=.45\textwidth]{Patient_16617_gray}
	\caption[Genetic distance versus time plot]{A plot of the genetic distance from the root versus time of the sequences from Patient 1180 from \cite{Llewellyn06} (LANL id: 16617). The dates of the internal nodes were estimated using \code{node.dating} and the internal nodes are included in the plot. The edges between adjacent nodes of the tree are drawn as solid lines and the molecular clock is repesented by a dashed line.}
	\label{fig:pat16617}
\end{figure}

\section{Future work} \label{sec:discuss}
One drawback of our methodology is that it assumes that the phylogeny follows a strict moelcular clock.
However, the local likelihood model can be extended to incoporate a variable moelcular clock; future work will include this extension.
The molecular clock assumption also implies that mutations are strictly additive over time, which is not true.
It may also be possible to (incorporate) this ``negative'' evolution into the model.

\section*{Acknowledgments} \label{sec:ackn}
We would like to thank Richard Liang for assistance in automating \emph{TempEst}.

\section*{Funding} \label{sec:fund}
This work was supported by the Canadian Institutes of Health Research (CIHR Team Grant: HIV Cure Research --- The Canadian HIV Cure Research Enterprise; CanCure) and by a CIHR operating grant to Art Poon (HOP-111406).

\bibliographystyle{natbib}
\bibliography{node.dating}

\end{document}
