\documentclass[12pt]{article}

\usepackage[margin=2cm]{geometry}
\usepackage{colortbl}
\usepackage{mathptmx}
\usepackage{nicefrac}
\usepackage{authblk}
\usepackage{array}

\usepackage{times}
\usepackage[round]{natbib}
\makeatletter
\renewcommand{\@biblabel}[1]{#1.}
\makeatother

\usepackage{graphicx}
\graphicspath{{./figures/}}

\usepackage{amsmath}
\usepackage{xcolor}

\newcommand{\code}[1]{{\tt #1}}
 
\begin{document}

\title{Supplementary Information \\ node.dating: : dating ancestors in phylogenetic trees in R}

\author[1,2,*]{Bradley R. Jones}
\author[2,3]{Art F.Y. Poon}
\affil[1]{Faculty of Health Sciences, Simon Fraser University, Burnaby, V5A 1S6, Canada}%, Burnaby, Canada}
\affil[2]{BC Centre for Excellence in HIV/AIDS, Vancouver, V6Z 1Y6, Canada}
\affil[3]{Department of Medicine, University of British Columbia, V5Z 1M9, Canada}%, Vancouver, Canada}
\affil[*]{Corresponding author (email: brj1@sfu.ca)}

\date{}

\maketitle

\section{Simulation} \label{sec:sim}
To verify the accuracy of \code{node.dating}, we applied it to simulated data.
We simulated 50 phylogenetic trees with 100 tips each using a birth-death model with the \emph{R} package \emph{TreeSim} \citep{TreeSim} using the parameters: $\lambda = 5.116 \times 10^{-2}$ day$^{-1}$, $\delta = 5.006 \times 10^{-2}$ day$^{-1}$, and $s = 5.237 \times 10^{-3}$.
We then applied a strict molecular clock to the trees with the \emph{R} package \emph{NELSI} \citep{NELSI} using the parameters: $\mu = \ 1.964\times 10^{-4}$ and $\sigma = \ 1.417\times 10^{-5}$ substitutions per generation.
We used these trees to simulate HIV sequences with \emph{INDELible} 1.03 \citep{Indelible09} using a HKY85 nucleotide substitution model \cite{HKY85} with a stationary distribution of 0.42, 0.15, 0.15, 0.28 for A, C, G, T respectively and a transitional bias of $\kappa = 8.5$.
Finally we reconstructed phylogenetic trees from the sequences using \emph{RAxML} version 8.2.4 \citep{Raxml14} with the GTR model and rooted the trees using the \code{rtt} function of the \emph{R} package, \emph{APE} \citep{APE}.
This process is engineered to replicate phylogenetic trees derived from real data.

\section{Weighted RMSE} \label{sec:rmse}
We couldn't iterate over the internal nodes to calculate an error metric because the tree topology may change after applying \emph{RAxML}.
Instead, we iterate over the pairs of tips in each tree and use the pair's most recent common ancestor (MRCA) in place of the internal node.
The dates of MRCA of each pair of tips of the original birth-death tree were saved and compared with the results of \code{node.dating} using a weighted root mean squared error (RMSE) as the error metric.
Specifically:
\[\operatorname{RMSE} = \sqrt{\frac{\sum_{1 \leq i < j \leq N}w_{i,j}\left(d_{\operatorname{MRCA}_{t_r}(i,j)} - \delta_{\operatorname{MRCA}_{t_p}(i,j)}\right)^2}{\sum_{1 \leq i < j \leq N}w_{i,j}}}\]
where $t_r$ and $t_p$ are the real (resp.~predicted) phylogenies each with $N$ tips; $\operatorname{MRCA}_t(i, j)$ is the MRCA of tip $i$ and $j$ in the phylogeny, $t$; $d_{m}$ and $\delta_m$ are the real (resp.~predicted) dates of the MRCA, $m$; and $w_{i, j}$ is the weight of the pair of tips, $i$ and $j$, and is given by:
\[w_{i, j} = \sqrt{\left(x_{\operatorname{MRCA}_{t_r}(i,j)}y_{\operatorname{MRCA}_{t_p}(i,j)}\right)^{-1}}\]
with $x_m$ and $y_m$ as the number of pairs of tips in the real (resp.~predicted) phylogeny whose MRCA is $m$.
Using these weights, the weighted RMSE is equal to the RMSE iterating over the internal nodes when the tree topologies are the same.
For our analysis we considered the mean of the RMSE of all 50 phylogenies.

\section{Acquisition of real data}
We retrieved intra-host patient-derived sequences from Patient 16617 on the LANL HIV database \\ (http://www.hiv.lanl.gov/, accessed June 24, 2015; Patient 1180 from \cite{Llewellyn06}).
Sequences were aligned using MUSCLE version 3.8.31 \cite{Muscle04} and inspected and cleaned using AliView \citep{AliView14}. 
We trimmed the alignments so that each sequence had at least 50\% coverage over each base.
We reconstructed the phylogeny of the patient's sequences using \emph{RAxML} version 8.2.4 \citep{Raxml14} and rooted the tree using the \code{rtt} function of APE \citep{APE}.

\bibliographystyle{bio}
\bibliography{node.dating}

\end{document}